### ISSUE TITLE
GenIdea/idea task should generate appropriate config from libs build.sc depends on 

### ISSUE THREAD

**Issue by Baccata** [lihaoyi/mill#347]
If the build depends on external libraries (imported through `import $ivy`), the GenIdea/idea task should generate configuration for those libraries accordingly. 

I assume the difficulty comes from the fact that the references to those libraries are not easily inspectable from the task. Maybe by looking at the classloader (ammonite has a SpecialClassLoader that keeps track of the urls of loaded jars) and inferring the various libs from where they are stored in the coursier cache ?

**Comment by lihaoyi** [comment]
Perhaps Mill could ask Ammonite what jars it has loaded into the classpath by the time the script is running, and use those? Ammonite already tracks all this stuff, and it should accurately reflect what code you are able to utilize in your `build.sc` and related scripts

**Comment by Baccata** [comment]
Is there an easy way to access ammonite's interpreter from mill ? My initial idea was : 

* get class loader 
* cross fingers and cast as SpecialClassLoader 
* call `allJars` to get jar paths 
* build library name from jar paths 

but if we can replace the `get class loader` bit by a safe reference, it'd be great 

Also, a quick look at the sources tells me that ammonite resolves the sources as well as the jars by default. Can you confirm ?

**Comment by lihaoyi** [comment]
For now just grab the SpecialClassloader however you can; once it's working we can put a nice API in place for Mill to use

Yeah ammonite resolves sources to make the `source` builtin utility work.

**Comment by lilyevsky** [comment]
This is still not working. In my build.sc I have this dependency:

import $ivy.`com.github.pureconfig::pureconfig:0.13.0`

The actual Mill build works fine, compiles, produces artifact, etc. It finds and downloads the pureconfig dependency.
However, mill.scalalib.GenIdea/idea task does not include it in the IDEA libraries, so I have to add it manually to avoid all that red stuff in the editor.
Saving the IDEA files in git after manual addition would not help; from time to time I need to run mill.scalalib.GenIdea/idea, and it will break everything again.

My environment: 
Windows or Linux (same problem on both). My workstation is behind corporate Nexus, in the predefScript.sc I override interp.repositories() to point to local maven repository and Nexus.

**Comment by lefou** [comment]
@lilyevsky  Can you please open a new issue with a repro?

**Comment by lilyevsky** [comment]
@lefou I started working on the new issue, and while doing that, I decided to do some more experiments.
Previously I did not tell you that in my build.sc the line

import $ivy.`com.github.pureconfig::pureconfig:0.13.0`

was surrounded by those special /*_*/ comments, like this:

/*_*/
import $ivy.`com.github.pureconfig::pureconfig:0.13.0`
/*_*/

The reason why I used those comments was, without them the whole line between backticks was in red.
With those comments, idea task did not add the pureconfig library, and so some other names in the code were in red and not recognizable by IntelliJ.

Now I tried to remove the comments, and the library was added. So I have the `com.github.pureconfig::pureconfig:0.13.0` in red, but everything else is fine.

Bottom line, this is not a problem. At least it is not the problem I thought it was.
The minor problem is IntelliJ not recognizing the import $ivy...... thing, but, I guess, mill cannot do anything about that.

Related question: do you know if anybody is planning to create a real mill Idea plugin? That would be really nice.
